includes/class-wc-payments.php

```diff
@@ -1551,6 +1557,8 @@ function ( $container ) {
 			}
 
 			new WooPay_Order_Status_Sync( self::$api_client, self::$account );
+		} else {
+			WooPay_Order_Status_Sync::remove_webhook();
```

Comment:
I wonder if this is the best place to remove the webhook, because this will trigger a separate query on every page load even if WooPay is simply disabled.

I don't have a better suggestion, but maybe we can remove it when the account cache is rewritten (maybe calling a separate hook in `Database_Cache`) and the conditions match (WooPay is no longer eligible), or make the webhook remove itself before it is run in case it no longer applies – account has been rejected or WooPay is no longer eligible.

Reply:
Sounds good. Let's move this to when the account is refreshed 👍

Reply:
done https://github.com/Automattic/woocommerce-payments/pull/8942/commits/86f585292901a8a7b6636f054f29f72f1fba5ff0